modules/mono/editor/GodotTools/GodotTools.IdeMessaging/Peer.cs

```diff
@@ -238,12 +238,12 @@ public async Task<TResponse> SendRequest<TResponse>(string id, string body)
                 queue.Enqueue(responseAwaiter);
             }
 
-            return (TResponse)await responseAwaiter;
+            return (TResponse?)await responseAwaiter;
```

Comment:
It doesn't look like `ResponseAwaiter` ever sets the result to null.

Reply:
It doesn't, but it stops the IDE from whining about converting to a possible null value, despite recognizing that `responseAwaiter` isn't null here

Reply:
It could be because you changed `GetResult` in `NotifyAwaiter` which `ResponseAwaiter` derives from: https://github.com/godotengine/godot/pull/87137#discussion_r1471668129

Reply:
That's exactly what's happening, yeah. I assumed that we wanted `GetResult` to be nullable though, as the result itself could be null and we don't have a safeguard against that. Would it be preferable to make it non-nullable & throw an exception if `result` is null?

Reply:
So something like:
```cs
public T GetResult() => result ?? throw exception ?? new ArgumentNullException(nameof(result));
```

Reply:
As I understand it, result will never be null if it was set from `SetResult`, only if the await completes with `SetException`. In this case, since exception is not null it will be thrown in `GetResult` already, so I'd say using the null forgiveness operator may be correct here.

```csharp
public T GetResult()
{
    if (exception != null)
        throw exception;
    return result!;
}
```